isvcs/zk.go

```diff
@@ -232,7 +233,72 @@ func zkFourLetterWord(server, command string, timeout time.Duration) ([]byte, er
 	return resp, nil
 }
 
-func BuildZookeeperInstance(instanceID int) dao.RunningService{
+type ZooKeeperServerStats struct {
+	InstanceID  int
+	Connections int
+	Mode        string
+}
+
+func GetAllZooKeeperServerStats() []ZooKeeperServerStats {
+	stats := []ZooKeeperServerStats{}
+	for _, key := range GetZooKeeperKeys() {
+		stats = append(stats, GetZooKeeperServerStats(key.InstanceID, key.Connection))
+	}
+
+	return stats
+}
+
+func GetZooKeeperServerStatsByID(instanceID int) ZooKeeperServerStats {
+	for _, key := range GetZooKeeperKeys() {
+		if key.InstanceID != instanceID {
+			continue
+		}
+
+		return GetZooKeeperServerStats(instanceID, key.Connection)
+	}
+	return ZooKeeperServerStats{}
+}
+
+func GetZooKeeperServerStats(instanceID int, connection string) ZooKeeperServerStats {
+	logger := log.
+		WithField("instanceid", instanceID).
+		WithField("connection", connection)
+
+	stats := ZooKeeperServerStats{InstanceID: instanceID}
+
+	bytes, err := zkFourLetterWord(connection, "srvr", 1*time.Second)
```

Comment:
A one second timeout is too short for a busy system. Please increase to 10 seconds like other 4-letter queries in this file.